File: stage1/main.py
import os
import sys

arg = sys.argv

print("The ARGUMENTS ARE: ", arg)

print(os.path.join('module', 'root_folder'))

if len(arg) == 1:
    print("Directory is not specified")
else:
    for root, dirs, files in os.walk(arg[1], topdown=True):
        for name in files:
            print(os.path.join(root, name))

File: stage1/main.go
package main

import (
	"fmt"
	"os"
	"path/filepath"
	"strings"
)

func main() {
	if len(os.Args) == 1 {
		fmt.Println("Directory is not specified")
	} else {
		dir := strings.Join(os.Args[1:], " ") // the directory is the second command line argument!
		// use the 'Walk' function to read 'dir' and print all the files
		err := filepath.Walk(dir, func(path string, info os.FileInfo, err error) error {
			if err != nil {
				return err
			}
			if info.IsDir() {
				return nil
			}
			fmt.Println(path)
			return nil
		})
		if err != nil {
			fmt.Println(err)
		}
	}
}
